### DISCUSSION TITLE
Codeql CPP mingw-w64 Compiller

### DISCUSSION THREAD

**Discussion by FaiqueAli** [github/codeql#10140] [Q&A] [answered]
Hi all,
I am trying to use codeql cli in my CPP project and trying to create a database file in a docker container.
but I am getting this issue after the execution of my build command and when codeql tries to generate a database

No source code was seen and extracted to C:\xxx.
This can occur if the specified build commands failed to compile or process any code.

For codebases written in Go, JavaScript, TypeScript, and Python, do not specify
an explicit --command.
For other languages, the --command must specify a "clean" build which compiles
all the source code files without reusing existing build artefacts.
I am using the GNU MinGW (v 12.2.0) compiler to compile my code.
The command  is being passed through an script(.bat) file

`cmake -G "MSYS Makefiles" ../
 make`

Does codeql support MinGW-w64 (v12.2.0) if not what is the soulution to generate codeql database?

**Comment by aibaars** [comment]
@FaiqueAli This seems to be a duplicate of https://github.com/github/codeql/issues/10132 . Would you like to continue the discussion here or on the other issue ticket?

**Reply by FaiqueAli** [reply, marked as answer]
I would like to continue at #10132.

**Comment by aibaars** [comment]
The CodeQL C/C++ analyser tracks which source files are compiled during the build process. The error message says that CodeQL somehow did not "see" any source files. This can have various causes:

CodeQL did not detect any compiler commands (such as gcc.exe)
CodeQL failed to parse the source files (or its included files)
Source files were found and parsed but they were not in the expected source location.
Could you have a look at codeql-database.yml in the database folder? There should be a property named sourceLocationPrefix.

Could you have a look in the src.zip file in the database folder? This file contains copies of the source files that CodeQL analysed. If there are files but the folder structure does not match the sourceLocationPrefix, then you could try to adjust the sourceLocationPrefix (-s, --source-root flag to codeql database create).

If there are few (or no) files in the zip file then CodeQL likely failed to detect the compiler calls. In this case you best have a look at the log/build-tracer.log file in the database folder.
